File: backend/src/main/java/com/ttwijang/cms/entity/Faq.java
package com.ttwijang.cms.entity;

import java.io.Serializable;
import java.time.LocalDateTime;

import javax.persistence.*;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.UpdateTimestamp;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "faq")
public class Faq implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(generator = "uuid2")
    @GenericGenerator(name = "uuid2", strategy = "uuid2")
    private String uid;

    @Column(nullable = false, length = 300)
    private String question;

    @Column(columnDefinition = "TEXT")
    private String answer;

    @Column(length = 50)
    private String category;

    @Column(columnDefinition = "INT DEFAULT 0")
    private Integer displayOrder;

    @CreationTimestamp
    private LocalDateTime createdDate;

    @UpdateTimestamp
    private LocalDateTime updatedDate;
}

File: client/src/api/faq.ts
import request from '@/utils/request';

export interface Faq {
  uid: string
  question: string
  answer: string
  category: string | null
  displayOrder: number
  createdDate: string
}

export function getFaqList() {
  return request({ url: '/faq', method: 'get' });
}

// 관리자 전용
export function createFaq(data: { question: string, answer?: string, category?: string, displayOrder?: number }) {
  return request({ url: '/admin/faq', method: 'post', data });
}

export function updateFaq(uid: string, data: { question?: string, answer?: string, category?: string, displayOrder?: number }) {
  return request({ url: `/admin/faq/${uid}`, method: 'put', data });
}

export function deleteFaq(uid: string) {
  return request({ url: `/admin/faq/${uid}`, method: 'delete' });
}
